From 08e036ee86703012f3b2d9b5f6b9b762b3ca42af Mon Sep 17 00:00:00 2001
From: Richard Lowe <richlowe@richlowe.net>
Date: Fri, 3 May 2019 01:20:30 +0000
Subject: [PATCH 17/39] i386: use correct error message about -msave-args and
 ABIs

We used to say it made no sense in 32bit mode, but it doesn't make sense
in 16bit mode either
---
 gcc/config/i386/i386-options.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/i386/i386-options.cc b/gcc/config/i386/i386-options.cc
index 920cad12f73..14c31b0b6c7 100644
--- a/gcc/config/i386/i386-options.cc
+++ b/gcc/config/i386/i386-options.cc
@@ -2776,7 +2776,7 @@ ix86_option_override_internal (bool main_args_p,
     }
 
   if (!TARGET_64BIT_P (opts->x_ix86_isa_flags) && TARGET_SAVE_ARGS)
-    error ("-msave-args makes no sense in the 32-bit mode");
+    error ("-msave-args only works in x32 or 64-bit mode");
 
   /* Validate -mpreferred-stack-boundary= value or default it to
      PREFERRED_STACK_BOUNDARY_DEFAULT.  */
-- 
2.53.0

